Gems/Perforce/Code/Source/PerforceSourceControl.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+
+#pragma once
+
+#include "IEditorClassFactory.h"
+#include "Include/IEditorClassFactory.h"
+
+#include "Include/ISourceControl.h"
+
+#include <QtCore/QString>
+
+class CPerforceSourceControl
+    : public ISourceControl
+    , public IClassDesc
+{
+public:
+    // constructor
+    CPerforceSourceControl() = default;
+    virtual ~CPerforceSourceControl() = default;
```

Comment:
we can probably do away with this abstraction in the editor code there is another interface that bridges this.

Reply:
Yes, the only interface that matters to O3DE and isn't legacy is hte source control busses.  Those are abstractions that represent "any source control currently plugged in".  the idea was that we could eventuallly have switchable (CVS/GIT/P4/etc).   And use the "local file SC" as the default fallback if nothing was plugged in so that somehting could at least complain about read-only file systems/files.

The interface in the O3DE source control bus is also agnostic wrt to the system underneath it, its structured as an INTENT based system

As in, "the user intends to edit this file - do what you need to do in order to make it editable and ready for the user" which could be a different set of operations depending on the source control provider.

It feels like the logical steps of this would be 

1. get something like this in, to seperate p4 out to make it not necessary to activate it
2. Update the core engine and editor code to only use the source control busses and helpers
3. mark this stuff as deprecated
4. Update the core editor/tooling code to auto install the local SC component if no source control is plugged into the interface after bootup.  Or alter it so that its some sort of priority stack.  Busses can have multiple ordered listeners, after all.
5. Move the perforce one into extras maybe, so you don't pay for compiling it every time you compile the engine without using it.